apps/pkcs12.c

```diff
@@ -555,7 +555,7 @@ int pkcs12_main(int argc, char **argv)
                 /* Look for matching private key */
                 for (i = 0; i < sk_X509_num(certs); i++) {
                     x = sk_X509_value(certs, i);
-                    if (X509_check_private_key(x, key)) {
+                    if (cert_matches_key(x, key)) {
```

Comment:
`cert_matches_key` will clear the error raised by X509_check_private_key, that may make sense...

Reply:
Yes, that was the purpose of this change.